### PR TITLE
onboarding improvements

### PR DESCRIPTION
**Pull request by SabrinaTardio**
Task/Issue URL: https://app.asana.com/0/1204186595873227/1208606247438571/f

**Description**: Improves some contextual onboarding edge cases and the dialog transition

**Steps to test this PR**:
1. Run the contextual onboarding and check the transition when doing first search and visit site happens after the site is loaded and it’s smooth.
2. Test Linear onboarding and check animations on fire and shield icon occur as expected
Test other scenarios

NonLinear1

go to debug menu → reset data → reset contextual onboarding
Navigate to [DuckDuckGo.com](https://duckduckgo.com/)
Visit a site (bbc.com) and check the tracker message appears
Should then continue as expected

NonLinear2

go to debug menu → reset data → reset contextual onboarding
Navigate to [DuckDuckGo.com](https://duckduckgo.com/)
Visit a site (google.com) and check the tracker message appears
Try and do a search -> search done dialog appears
Visit a site (bbc.com) and check the tracker message appears again (since before was a major tracker)

NonLinear3

go to debug menu → reset data → reset contextual onboarding
Navigate to [DuckDuckGo.com](https://duckduckgo.com/)
Visit a site (bbc.com) and check the tracker message appears
Try and do a search -> search done dialog appears
Visit a site (google.com) and the fire button message should appear (since before we blocked trackers)

NonLinear4

go to debug menu → reset data → reset contextual onboarding
Navigate to [DuckDuckGo.com](https://duckduckgo.com/)
Do a non suggested search
should then continue like the linear

NonLinear 5
go to debug menu → reset data → reset contextual onboarding
Navigate to [DuckDuckGo.com](https://duckduckgo.com/)
Get to the try a site dialog
perform another search
the dialog should disappear

NonLinear 6
go to debug menu → reset data → reset contextual onboarding
Navigate to [DuckDuckGo.com](https://duckduckgo.com/)
Follow linear onboarding until you visit your first site
Perform a search
You should see the try fire button dialog

Try other scenarios you like

<!—
Tagging instructions
If this PR isn't ready to be merged for whatever reason it should be marked with the `DO NOT MERGE` label (particularly if it's a draft)
If it's pending Product Review/PFR, please add the `Pending Product Review` label.

If at any point it isn't actively being worked on/ready for review/otherwise moving forward (besides the above PR/PFR exception) strongly consider closing it (or not opening it in the first place). If you decide not to close it, make sure it's labelled to make it clear the PRs state and comment with more information.
—>

**Definition of Done**:

* [ ] Does this PR satisfy our [Definition of Done](https://app.asana.com/0/1202500774821704/1207634633537039/f)?

—
###### Internal references:
[Pull Request Review Checklist](https://app.asana.com/0/1202500774821704/1203764234894239/f)
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
[Pull Request Documentation](https://app.asana.com/0/1202500774821704/1204012835277482/f)

### CODE REVIEW

**Comment by alessandroboron** [comment]
Non Linear 2
- I think the App behaves as expected but the test instructions are wrong:
   - Try and do a search -> no dialog should appear. This should appear if we haven’t done a search before.

Non Linear 3
- Same as Non Linear 2 comment

File: DuckDuckGo/Onboarding/ContextualOnboarding/ContextualOnboardingStateMachine.swift [hunk 1]
```diff
@@ -136,7 +151,7 @@ final class ContextualOnboardingStateMachine: ContextualOnboardingDialogTypeProv
 
     private weak var lastVisitTab: Tab?
     private var lastVisitSite: URL?
-    private var notBlockedTrackerSeen: Bool = false
+    var notBlockedTrackerSeen: Bool = false
```
Comment (alessandroboron) [inline DuckDuckGo/Onboarding/ContextualOnboarding/ContextualOnboardingStateMachine.swift:154]: not sure how I feel about having this exposed only for testing purpose.

By any chance, could we call `updateStateFor(tab:)` and let the state machine set that without exposing it?

**Comment by SabrinaTardio** [comment]
> Non Linear 2
> 
> * I think the App behaves as expected but the test instructions are wrong:
>   
>   * Try and do a search -> no dialog should appear. This should appear if we haven’t done a search before.
> 
> Non Linear 3
> 
> * Same as Non Linear 2 comment

Yes that is correct I updated the test cases

File: UnitTests/Onboarding/ContextualOnboarding/ContextualOnboardingStateMachineTests.swift [hunk 2]
```diff
@@ -315,22 +320,340 @@ class ContextualOnboardingStateMachineTests: XCTestCase {
         }
     }
 
+    func test_UpdateStateFor_OpensDuckDuckGoSearch_ShowsTryASearch() {
```
Comment (alessandroboron) [inline UnitTests/Onboarding/ContextualOnboarding/ContextualOnboardingStateMachineTests.swift:323]: Should we keep the naming of these tests consistent with the rest of the tests already written? 

e.g. test_UpdateStateFor_When<condition>_Then<whathappens>

**Review by alessandroboron — APPROVED** [review]
LGTM! I left a couple of comments.
